opsdroid/cli/utils.py

```diff
@@ -160,3 +162,46 @@ def welcome_message(config):
         _LOGGER.warning(
             _("'welcome-message: true/false' is missing in configuration.yaml")
         )
+
+
+def list_modules(ctx, param, value):
+    """List the active modules from confiv."""
+    config = load_config_file(
+        ["configuration.yaml", DEFAULT_CONFIG_PATH, "/etc/opsdroid/configuration.yaml"]
+    )
+
+    try:
+        if param == "connectors":
+            connectors = [connector for connector in config.get("connectors")]
+            click.echo(
+                "Connectors active in the configuration: {connectors_active}.".format(
+                    connectors_active=", ".join(connectors)
+                )
+            )
+
+        if param == "databases":
+            databases = [database for database in config.get("databases")]
+            click.echo(
+                "Databases active in the configuration: {databases_active}.".format(
+                    databases_active=", ".join(databases)
+                )
+            )
+
+        if param == "skills":
+            skills = [skill for skill in config.get("skills")]
+            click.echo(
+                "Skills active in the configuration: {skills_active}.".format(
+                    skills_active=", ".join(skills)
+                )
+            )
+
+        if param == "parsers":
+            parsers = [parser for parser in config.get("parsers")]
+            click.echo(
+                "Parsers active in the configuration: {parsers_active}.".format(
+                    parsers_active=", ".join(parsers)
+                )
+            )
```

Comment:
You could condense all this down to `things = [thing for thing in config.get(param)] `. It would also be nice to get some more information here in a tabular format

We should echo each module on a separate line. For example if I have the `hello`, `dance` and a custom skill configured it would be nice to get some output like:

```
NAME     TYPE   MODE  CACHED  LOCATION
hello    skill  repo  yes     https://github.com/opsdroid/skill-hello.git
dance    skill  repo  yes     https://github.com/opsdroid/skill-dance.git
myskill  skill  path  no      /path/to/my/skill
```

Or if you have the slack connector configured you would get


```
NAME     TYPE       MODE    CACHED  LOCATION
slack    connector  module  yes     opsdroid.connector.slack
```

This kind of layout could make it nice to operate on from the command line with tools like `grep` and `xargs`.

Reply:
I thought this would be something easy to implement but came around and bit me.  Perhaps I should revert these changes a create a new PR for this 🤔 

I'm thinking that perhaps a way to do this would be to actually load the config and check if its cached and the location. What do you think?